tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +686,205 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+		{
+			var project = "AppWithXCAssets";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+			var project_path = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out var appPath);
+
+			ConfigureAssets (project_path, runtimeIdentifiers, startWithAssets);
+
+			var appExecutable = GetNativeExecutable (platform, appPath);
+			ExecuteWithMagicWordAndAssert (platform, runtimeIdentifiers, appExecutable);
+
+			var resourcesDirectory = GetResourcesDirectory (platform, appPath);
+
+			var assetsCar = Path.Combine (resourcesDirectory, "Assets.car");
+			Assert.That (assetsCar, Does.Exist, "Assets.car");
+
+			var doc = ProcessAssets (assetsCar, GetFullSdkVersion (sdkVersion));
+			Assert.IsNotNull (doc, "There was an issue processing the asset binary.");
+
+			var foundAssets = FindAssets (doc);
+
+			// Seems the 2 vectors are not being consumed in MacCatalyst but they still appear in the image Datasets
+			var TotalUniqueAssets = platform == ApplePlatform.MacCatalyst ? 14 : 16;
+
+			Assert.AreEqual (TotalUniqueAssets, foundAssets.Count, "Wrong number of assets found");
+			Assert.IsFalse (foundAssets.Contains ("Data.DS_StoreDataTest"), "DS_Store files should not be included.");
+
+			var arm64txt = Path.Combine (resourcesDirectory, "arm64.txt");
+			var armtxt = Path.Combine (resourcesDirectory, "arm.txt");
+			var x64txt = Path.Combine (resourcesDirectory, "x64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm64")), File.Exists (arm64txt), "arm64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm")), File.Exists (armtxt), "arm.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-x64")), File.Exists (x64txt), "x64.txt");
+		}
+
+		void ConfigureAssets (string project_path, string runtimeIdentifiers, bool startWithAssets)
+		{
+			// We either want the assets added before the build, or we will be adding them after the build
+			if (startWithAssets)
+				CopyAssets (project_path);
+			else
+				DeleteAssets (project_path);
+
+			Clean (project_path);
+
+			DotNet.AssertBuild (project_path, GetDefaultProperties (runtimeIdentifiers));
+
+			if (!startWithAssets) {
+				CopyAssets (project_path);
+				// Building the project twice without cleaning in between fails: https://github.com/xamarin/maccore/issues/2530
+				Clean (project_path);
+				DotNet.AssertBuild (project_path, GetDefaultProperties (runtimeIdentifiers));
+			}
+		}
+
+		void DeleteAssets (string project_path)
+		{
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+			try {
+				if (Directory.Exists (xcassetsDir))
+					Directory.Delete (xcassetsDir, true);
+			}
+			catch (Exception e){
+				Assert.Fail ($"Deleting Assets.xcassets failed: {e.Message}");
+			}
+		}
+
+		void CopyAssets (string project_path)
+		{
+			DirectoryInfo testingAssetsDir = new DirectoryInfo (Path.Combine (project_path, @"../../TestingAssets"));
```

Comment:
Why the need of a verbatim string literal?

Reply:
Ah I just used it somewhere else and used it here too. I'll remove it!